src/components/Species/AddSpeciesModal.tsx

```diff
@@ -339,6 +346,39 @@ export default function AddSpeciesModal(props: AddSpeciesModalProps): JSX.Elemen
             }
           />
         </Grid>
+        <Grid item xs={12}>
+          <MultiSelect
+            fullWidth={true}
+            label={strings.ECOSYSTEM_TYPE}
+            tooltipTitle={
+              <>
+                {`${strings.TOOLTIP_ECOSYSTEM_TYPE} `}
+                <a
+                  target='_blank'
+                  rel='noopener noreferrer'
+                  href='https://www.worldwildlife.org/publications/terrestrial-ecoregions-of-the-world'
+                >
+                  {strings.LEARN_MORE}
+                </a>
+              </>
+            }
+            onAdd={(type: EcosystemType) => {
+              const selectedTypes = record.ecosystemTypes ?? [];
+              selectedTypes.push(type);
+              onChange('ecosystemTypes', selectedTypes);
+            }}
+            onRemove={(type: EcosystemType) => {
+              const selectedTypes = record.ecosystemTypes ?? [];
+              const index = selectedTypes.findIndex((x) => x === type);
+              selectedTypes.splice(index, 1);
+              onChange('ecosystemTypes', selectedTypes);
```

Comment:
Above code works.. in case it helps  functional constructs for readability/brevity ..
‘’’
onChange(‘ecosystemType’, record.ecosystemTypes?.filter(ecosystemType => ecosystemType !== type) ?? []);
‘’’
Can’t find the back tick on my phone

Reply:
I can do that (I also see that I forgot to check whether index === -1)